packages/core/src/external/carequality/iti55/process-incoming-iti-55.ts

```diff
@@ -0,0 +1,205 @@
+import { AxiosResponse } from "axios";
+import {
+  PatientDiscoveryRequestIncoming,
+  PatientDiscoveryResponseOutgoing,
+} from "@metriport/ihe-gateway-sdk";
+import { PatientDataMPI, convertPatientToFHIR } from "../../mpi/patient-incoming-schema";
+import {
+  validateFHIRAndExtractPatient,
+  InternalError,
+  PatientAddressRequestedError,
+  LivingSubjectAdministrativeGenderRequestedError,
+} from "./validating-iti55";
+import { matchPatients, jaroWinklerSimilarity } from "../../mpi/match-patients";
+import { normalizePatientDataMPI } from "../../mpi/normalize-patient";
+import { mergePatients, mergeWithFirstPatient } from "../../mpi/merge-patients";
+import { capture } from "../../../util/notifications";
+import { getEnvVarOrFail } from "../../../util/env-var";
+
+import axios from "axios";
+const ossApi = axios.create();
+const SIMILARITY_THRESHOLD = 0.96;
+const METRIPORT_HOME_COMMUNITY_ID = "urn:oid:2.16.840.1.113883.3.9621";
+
+type MPIBlockParams = {
+  dob: string;
+  genderAtBirth: string;
+  firstNameInitial?: string;
+  lastNameInitial?: string;
+};
+
+function constructErrorResponse(
+  payload: PatientDiscoveryRequestIncoming,
+  codingSystem: string,
+  code: string,
+  error: string,
+  patientMatch: boolean | null = null
+): PatientDiscoveryResponseOutgoing {
+  return {
+    id: payload.id,
+    timestamp: payload.timestamp,
+    responseTimestamp: new Date().toISOString(),
+    patientMatch: patientMatch,
+    xcpdHomeCommunityId: METRIPORT_HOME_COMMUNITY_ID,
```

Comment:
we normally try to use undefined and not null, but iti55 specs require null if there is an internal server error